homeassistant/components/generic/config_flow.py

```diff
@@ -168,10 +168,15 @@ async def async_test_still(hass, info) -> tuple[dict[str, str], str | None]:
     return {}, f"image/{fmt}"
 
 
-def slug_url(url) -> str | None:
+def slug_url(hass, url) -> str | None:
     """Convert a camera url into a string suitable for a camera name."""
     if not url:
         return None
+    if not isinstance(url, template_helper.Template):
+        url = cv.template(url)
+        url.hass = hass
```

Comment:
This can raise `vol.Invalid`. Prefer to instantiate it directly:
```suggestion
        url = template_helper.Template(url, hass)
```

Reply:
done [b1ca479](https://github.com/home-assistant/core/pull/71031/commits/b1ca47953e64f5b2afbc08cf310d1954dafaf7dd)